pkg/asset/manifests/azure/cluster.go

```diff
@@ -240,3 +253,32 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 		},
 	}, nil
 }
+
+func getNextAvailableIP(ctx context.Context, installConfig *installconfig.InstallConfig) (string, error) {
+	lbip := "10.0.0.100"
+	client, err := installConfig.Azure.Client()
+	if err != nil {
+		return "", fmt.Errorf("failed to get azure client: %w", err)
+	}
+	vClient, err := client.GetVirtualNetworksClient(ctx)
+	if err != nil {
+		return "", fmt.Errorf("failed to get azure virtual network client: %w", err)
+	}
+	if installConfig.Config.Azure.InternalLBIP != "" {
+		lbip = installConfig.Config.Azure.InternalLBIP
+	}
```

Comment:
As I mentioned in the other comment, I don't think we need to let users be able to specify this value. But I do think we need to make sure that the LB IP address is within the machineCIDR range. I'm not certain, but I'm pretty sure that is true. 

So I think the overall logic should be something like:
- check if capz default ip is within machineCIDR
- if so, check that it is available as you are already doing
- if not, check if the first ip address in the cidr range is available
- then use that or the next available as you are already doing